File: Snooper/src/androidTest/java/com/prateekj/snooper/formatter/JsonResponseFormatterTest.java
package com.prateekj.snooper.formatter;


import org.junit.Test;

import static com.prateekj.snooper.utils.TestUtilities.readFrom;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class JsonResponseFormatterTest {

  @Test
  public void shouldReturnFormattedJsonObject() throws Exception {
    ResponseFormatter formatter = new JsonResponseFormatter();
    String formattedResponse = formatter.format(readFrom("person_details_raw_response.json"));
    String expectedResponse = readFrom("person_details_formatted_response.json");

    assertThat(formattedResponse, is(expectedResponse));
  }

  @Test
  public void shouldReturnFormattedJsonArray() throws Exception {
    ResponseFormatter formatter = new JsonResponseFormatter();
    String formattedResponse = formatter.format(readFrom("person_details_raw_array_response.json"));
    String expectedResponse = readFrom("person_details_formatted_array_response.json");

    assertThat(formattedResponse, is(expectedResponse));
  }

  @Test
  public void shouldReturnSameResponseWhenExceptionOccurs() throws Exception {
    ResponseFormatter formatter = new JsonResponseFormatter();
    String invalidJSON = "{\"value\":\"Invalid JSON\", \"value\"}";
    String formattedResponse = formatter.format(invalidJSON);

    assertThat(formattedResponse, is(invalidJSON));
  }

}

File: Snooper/src/androidTest/java/com/prateekj/snooper/formatter/JsonResponseFormatterTest.kt
package com.prateekj.snooper.formatter


import com.prateekj.snooper.utils.TestUtilities.readFrom
import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat
import org.junit.Test

class JsonResponseFormatterTest {

  @Test
  @Throws(Exception::class)
  fun shouldReturnFormattedJsonObject() {
    val formatter = JsonResponseFormatter()
    val formattedResponse = formatter.format(readFrom("person_details_raw_response.json"))
    val expectedResponse = readFrom("person_details_formatted_response.json")

    assertThat(formattedResponse, `is`<String>(expectedResponse))
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnFormattedJsonArray() {
    val formatter = JsonResponseFormatter()
    val formattedResponse = formatter.format(readFrom("person_details_raw_array_response.json"))
    val expectedResponse = readFrom("person_details_formatted_array_response.json")

    assertThat(formattedResponse, `is`<String>(expectedResponse))
  }

  @Test
  @Throws(Exception::class)
  fun shouldReturnSameResponseWhenExceptionOccurs() {
    val formatter = JsonResponseFormatter()
    val invalidJSON = "{\"value\":\"Invalid JSON\", \"value\"}"
    val formattedResponse = formatter.format(invalidJSON)

    assertThat(formattedResponse, `is`(invalidJSON))
  }

}